src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.netcoreapp.cs

```diff
@@ -225,7 +226,118 @@ public static void SignCmsUsingECDsaCertWithNotMatchingKeyThrows()
                 Assert.Throws<CryptographicException>(() => cms.ComputeSignature(signer));
             }
         }
-        
+
+        [Fact]
+        public static void AddCertificate()
+        {
+            SignedCms cms = new SignedCms();
+            cms.Decode(SignedDocuments.CounterSignedRsaPkcs1OneSigner);
+
+            int numOfCerts = cms.Certificates.Count;
+
+            X509Certificate2 newCert = Certificates.RSAKeyTransfer1.GetCertificate();
+            cms.AddCertificate(newCert);
+
+            Assert.Equal(numOfCerts + 1, cms.Certificates.Count);
+
+            Assert.Equal(1, cms.Certificates.OfType<X509Certificate2>().Where((cert) => cert.Equals(newCert)).Count());
+
+            cms.CheckSignature(true);
+        }
+
```

Comment:
Also useful: Adding/removing certs do not alter existing returned collections from cms.Certificates.
Maybe also useful: Removing matches when "toRemove" has a private key bound.